tests/contracts-solidity.spec.ts

```diff
@@ -98,18 +96,90 @@ describe("Solang Smart Contracts", () => {
     );
     expect(initialValue).toBeDefined();
     expect(initialValue.toString()).toEqual("0x01");
-    
-    // Call contract with Action: 0xCDE4EFA9 = Action::Flip()
-    await callContract(api, contractCreator, address, flipAction);
+
+    await callContract(api, contractCreator, address, abi.messages.flip());
 
     const newValue = await getContractStorage(api, address, STORAGE_KEY);
     expect(newValue.toString()).toEqual("0x00");
 
-    await callContract(api, contractCreator, address, flipAction);
+    let res = await rpcContract(api, address, abi.messages.get());
+    expect(res.toString()).toEqual("0x00");
+
+    await callContract(api, contractCreator, address, abi.messages.flip());
 
     const flipBack = await getContractStorage(api, address, STORAGE_KEY);
     expect(flipBack.toString()).toEqual("0x01");
 
+    res = await rpcContract(api, address, abi.messages.get());
+    expect(res.toString()).toEqual("0x01");
+
+    done();
+  });
+
+  test("Raw Creator contract", async (done): Promise<void> => {
+    // The next two lines are a not so pretty workaround until the new metadata format has been fully implemented
```

Comment:
If `const abi = new Abi()` is working now, this line can be deleted :)

Reply:
Removed